hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/ispn/IspnResource.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.service.ispn;
+
+import java.io.Serializable;
+
+import org.hawkular.inventory.model.Metric;
+import org.hawkular.inventory.model.Resource;
+import org.hibernate.search.annotations.Analyze;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.annotations.Store;
+
+/**
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Indexed(index = "resource")
+public class IspnResource implements Serializable {
+
+    @Field(store = Store.YES, analyze = Analyze.NO, indexNullAs = Field.DEFAULT_NULL_TOKEN)
+    private final String feedId;
+
+    @Field(store = Store.YES, analyze = Analyze.NO, indexNullAs = Field.DEFAULT_NULL_TOKEN)
+    private final String typeId;
+
+    @Field(store = Store.YES, analyze = Analyze.NO, indexNullAs = Field.DEFAULT_NULL_TOKEN)
+    private final String parentId;
+
+    private final Resource resource;
```

Comment:
I'm not sure if that's ok. If the whole stateful "Resource" object is serialized with IspnResource, then we may have the same side-effects that I hit before, with lazy-loaded "children" being serialized as well here? The mutability of "resource" is going to leak to IspnResource.

Reply:
Oh, no because it's cloned, I understand.
But i'm still in favor of Immutable "Resource" :)

Reply:
Thinking again about it... it's cloned at construction, but then it's provided through getter without re-cloning, so it can still be mutated. I think it's a problem.